### ISSUE TITLE
CyberPanel installer not setting hostname for postfix hostname causing service failure

### ISSUE THREAD

**Issue by meramsey** [litespeedtech/cyberpanel#587]
So after some review based on cyberpanel slack report was able to confirm after a fresh install of cyberpanel on ubuntu and on centos that neither has the postfix set properly with the hostname.

```
Welcome to Ubuntu 20.04.2 LTS (GNU/Linux 5.4.0-72-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Fri Apr 16 21:48:37 UTC 2021

  System load:  0.42              Processes:             150
  Usage of /:   1.7% of 96.75GB   Users logged in:       0
  Memory usage: 2%                IPv4 address for eth0: 93.158.203.101
  Swap usage:   0%                IPv6 address for eth0: 2a00:1ca8:2a:9::19


0 updates can be installed immediately.
0 of these updates are security updates.


Last login: Fri Apr 16 21:43:53 2021 from 144.202.38.159
root@cyberpanel:~# hostname
cyberpanel.whattheserver.com
root@cyberpanel:~# sh <(curl https://cyberpanel.net/install.sh || wget -O - https://cyberpanel.net/install.sh)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1305  100  1305    0     0  16730      0 --:--:-- --:--:-- --:--:-- 16730
wget is already the newest version (1.20.3-1ubuntu1).
curl is already the newest version (7.68.0-1ubuntu2.5).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

Initializing...

Fetching latest data from CyberPanel server...

This may take few seconds...

Checking root privileges...

You are runing as root...

Valid IP detected...

Checking server location...

System: Ubuntu 20 detected...

Checking virtualization type...

Initializing...

                CyberPanel Installer v2.1.1

1. Install CyberPanel.

2. Addons and Miscellaneous

3. Exit.


  Please enter the number[1-3]: 1

                CyberPanel Installer v2.1.1

RAM check : 152/7961MB (1.91%)

Disk check : 1/97GB (2%) (Minimal 10GB free space)

1. Install CyberPanel with OpenLiteSpeed.

2. Install Cyberpanel with LiteSpeed Enterprise.

3. Exit.


  Please enter the number[1-3]: 1


Install Full service for CyberPanel? This will include PowerDNS, Postfix and Pure-FTPd.

Full installation [Y/n]: y

Full installation selected...

Do you want to setup Remote MySQL? (This will skip installation of local MySQL)

(Default = No) Remote MySQL [y/N]: n

Local MySQL selected...

Press Enter key to continue with latest version or Enter specific version such as: 1.9.4 , 2.0.1 , 2.0.2 ...etc
2.1.1

Set branch name to v2.1.1...

Please choose to use default admin password 1234567, randomly generate one (recommended) or specify the admin password?
Choose [d]fault, [r]andom or [s]et password: [d/r/s] r

Admin password will be provided once installation is completed...


Do you wish to install Memcached process and its PHP extension?
Please select [Y/n]: y

Install Memcached process and its PHP extension set to Yes...


Do you wish to install Redis process and its PHP extension?
Please select [Y/n]: n

Would you like to set up a WatchDog (beta) for Web service and Database service ?
The watchdog script will be automatically started up after installation and server reboot
If you want to kill the watchdog , run watchdog kill
Please type Yes or no (with capital Y, default Yes): 
no
Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done



              Website : https://www.cyberpanel.net                 
              Forums  : https://forums.cyberpanel.net              
              Wikipage: https://docs.cyberpanel.net                
                                                                   
            Enjoy your accelerated Internet by                  
                CyberPanel & OpenLiteSpeed                                                   
###################################################################
If your provider has a network-level firewall
Please make sure you have opened following port for both in/out:
TCP: 8090 for CyberPanel
TCP: 80, TCP: 443 and UDP: 443 for webserver
TCP: 21 and TCP: 40110-40210 for FTP
TCP: 25, TCP: 587, TCP: 465, TCP: 110, TCP: 143 and TCP: 993 for mail service
TCP: 53 and UDP: 53 for DNS service
Would you like to restart your server now? [y/N]: n
root@cyberpanel:~# postconf| grep -v warning| grep 'myhostname = '
myhostname = server.example.com
root@cyberpanel:~# hostname
cyberpanel.whattheserver.com
root@cyberpanel:~#
```

I explicitly terminated VPS reinstalled fresh ubuntu server set hostname and checked it then before installing cyberpanel and also after cyberpanel and its set but the logic for the setting hostname in installer is not working.

I think it has to do with this:
https://github.com/usmannasir/cyberpanel/blob/stable/install/install.py#L746-L834

Some other possible things could also be failing and related
https://github.com/usmannasir/cyberpanel/issues/585

**Comment by meramsey** [comment]
What's even weirder is that when i looked through the logs it shows like its set but never is in the config even though service like seems to think it is


```
[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] Running: apt-get -y debconf-utils

[04.16.2021_22-02-50] #########################################################################

E: Invalid operation debconf-utils



[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] Running apt-get -y debconf-utils failed. Running again, try number 1

[04.16.2021_22-02-50] #########################################################################

E: Invalid operation debconf-utils



[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] Running apt-get -y debconf-utils failed. Running again, try number 2

[04.16.2021_22-02-50] #########################################################################

E: Invalid operation debconf-utils



[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] Running apt-get -y debconf-utils failed. Running again, try number 3

[04.16.2021_22-02-50] #########################################################################




[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] [ERROR] We are not able to run apt-get -y debconf-utils return code: 100.

[04.16.2021_22-02-50] #########################################################################




[04.16.2021_22-02-50] #########################################################################

[04.16.2021_22-02-50] Running: debconf-set-selections /root/cyberpanel/install/pf.unattend.text

[04.16.2021_22-02-50] #########################################################################




[04.16.2021_22-02-51] #########################################################################

[04.16.2021_22-02-51] Successfully ran: debconf-set-selections /root/cyberpanel/install/pf.unattend.text.

[04.16.2021_22-02-51] #########################################################################




[04.16.2021_22-02-51] #########################################################################

[04.16.2021_22-02-51] Running: apt-get -y install postfix postfix-mysql

[04.16.2021_22-02-51] #########################################################################

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  ssl-cert
Suggested packages:
  procmail postfix-pgsql postfix-ldap postfix-pcre postfix-lmdb postfix-sqlite sasl2-bin | dovecot-common resolvconf postfix-cdb mail-reader
  postfix-doc openssl-blacklist
The following NEW packages will be installed:
  postfix postfix-mysql ssl-cert
0 upgraded, 3 newly installed, 0 to remove and 0 not upgraded.
Need to get 1,236 kB of archives.
After this operation, 4,734 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal/main amd64 ssl-cert all 1.0.39 [17.0 kB]
Get:2 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postfix amd64 3.4.13-0ubuntu1 [1,198 kB]
Get:3 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 postfix-mysql amd64 3.4.13-0ubuntu1 [21.5 kB]
Fetched 1,236 kB in 0s (12.5 MB/s)  
Preconfiguring packages ...
Can't exec "/tmp/ssl-cert.config.yZtzYD": Permission denied at /usr/share/perl/5.30/IPC/Open3.pm line 281.
open2: exec of /tmp/ssl-cert.config.yZtzYD configure  failed: Permission denied at /usr/share/perl5/Debconf/ConfModule.pm line 59.
Selecting previously unselected package ssl-cert.
(Reading database ... 109138 files and directories currently installed.)
Preparing to unpack .../ssl-cert_1.0.39_all.deb ...
Unpacking ssl-cert (1.0.39) ...
Selecting previously unselected package postfix.
Preparing to unpack .../postfix_3.4.13-0ubuntu1_amd64.deb ...
Unpacking postfix (3.4.13-0ubuntu1) ...
Selecting previously unselected package postfix-mysql.
Preparing to unpack .../postfix-mysql_3.4.13-0ubuntu1_amd64.deb ...
Unpacking postfix-mysql (3.4.13-0ubuntu1) ...
Setting up ssl-cert (1.0.39) ...
Setting up postfix (3.4.13-0ubuntu1) ...
Adding group `postfix' (GID 122) ...
Done.
Adding system user `postfix' (UID 114) ...
Adding new user `postfix' (UID 114) with group `postfix' ...
Not creating home directory `/var/spool/postfix'.
Creating /etc/postfix/dynamicmaps.cf
Adding group `postdrop' (GID 123) ...
Done.
setting myhostname: cyberpanel.whattheserver.com
setting alias maps
setting alias database
changing /etc/mailname to cyberpanel.whattheserver.com
setting myorigin
setting destinations: $myhostname, cyberpanel.whattheserver.com, localhost.whattheserver.com, localhost
setting relayhost: 
setting mynetworks: 127.0.0.0/8 [::ffff:127.0.0.0]/104 [::1]/128
setting mailbox_size_limit: 0
setting recipient_delimiter: +
setting inet_interfaces: all
setting inet_protocols: all
/etc/aliases does not exist, creating it.
WARNING: /etc/aliases exists, but does not have a root alias.

Postfix (main.cf) is now set up with a default configuration.  If you need to 
make changes, edit /etc/postfix/main.cf (and others) as needed.  To view 
Postfix configuration values, see postconf(1).

After modifying main.cf, be sure to run 'systemctl reload postfix'.
```





============

File shows not set.


```
root@cyberpanel:~# cat /etc/postfix/main.cf
# Global Postfix configuration file. This file lists only a subset
# of all parameters. For the syntax, and for a complete parameter
# list, see the postconf(5) manual page (command: "man 5 postconf").
#
# For common configuration examples, see BASIC_CONFIGURATION_README
# and STANDARD_CONFIGURATION_README. To find these documents, use
# the command "postconf html_directory readme_directory", or go to
# http://www.postfix.org/.
#
# For best results, change no more than 2-3 parameters at a time,
# and test if Postfix still works after every change.



queue_directory = /var/spool/postfix
command_directory = /usr/sbin
daemon_directory = /usr/lib/postfix/sbin
data_directory = /var/lib/postfix
mail_owner = postfix
inet_protocols = all
mydestination = localhost, localhost.localdomain
unknown_local_recipient_reject_code = 550
alias_maps = hash:/etc/aliases
alias_database = hash:/etc/aliases
debug_peer_level = 2
debugger_command =
         PATH=/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin
         ddd $daemon_directory/$process_name $process_id & sleep 5

sendmail_path = /usr/sbin/sendmail.postfix
newaliases_path = /usr/bin/newaliases.postfix
mailq_path = /usr/bin/mailq.postfix
setgid_group = postdrop
html_directory = no
manpage_directory = /usr/share/man
sample_directory = /usr/share/doc/postfix-2.10.1/samples
readme_directory = /usr/share/doc/postfix-2.10.1/README_FILES


myhostname = server.example.com
mynetworks = 127.0.0.0/8
message_size_limit = 30720000
virtual_alias_domains = 
virtual_alias_maps = proxy:mysql:/etc/postfix/mysql-virtual_forwardings.cf, mysql:/etc/postfix/mysql-virtual_email2email.cf
virtual_mailbox_domains = proxy:mysql:/etc/postfix/mysql-virtual_domains.cf
virtual_mailbox_maps = proxy:mysql:/etc/postfix/mysql-virtual_mailboxes.cf
virtual_mailbox_base = /home/vmail
virtual_uid_maps = static:5000
virtual_gid_maps = static:5000
smtpd_sasl_type = dovecot
smtpd_sasl_path = private/auth
smtpd_sasl_auth_enable = yes
broken_sasl_auth_clients = yes
smtpd_sasl_authenticated_header = yes
smtpd_recipient_restrictions = permit_mynetworks, permit_sasl_authenticated, reject_unauth_destination
smtpd_use_tls = yes
smtpd_tls_cert_file = /etc/pki/dovecot/certs/dovecot.pem
smtpd_tls_key_file = /etc/pki/dovecot/private/dovecot.pem
virtual_create_maildirsize = yes
virtual_maildir_extended = yes
proxy_read_maps = $local_recipient_maps $mydestination $virtual_alias_maps $virtual_alias_domains $virtual_mailbox_maps $virtual_mailbox_domains $relay_recipient_maps $relay_domains $canonical_maps $sender_canonical_maps $recipient_canonical_maps $relocated_maps $transport_maps $mynetworks $virtual_mailbox_limit_maps
virtual_transport = dovecot
dovecot_destination_recipient_limit = 1
inet_interfaces = all
smtp_tls_security_level = may

smtpd_milters = inet:127.0.0.1:8891
non_smtpd_milters = $smtpd_milters
milter_default_action = accept
root@cyberpanel:~#
```

Looks like it only really updated.

```
root@cyberpanel:~# grep -rn cyberpanel.whattheserver.com /etc
/etc/mailname:1:cyberpanel.whattheserver.com
Binary file /etc/aliases.db matches
/etc/hostname:1:cyberpanel.whattheserver.com
/etc/hosts:8:127.0.1.1 cyberpanel.whattheserver.com cyberpanel
/etc/hosts:9:127.0.0.1 localhost cyberpanel.whattheserver.com
root@cyberpanel:~#
```

@usmannasir @qtwrk   any ideas or thoughts now im more unsure lol

**Comment by meramsey** [comment]
I did this which i think should fix the issue.
https://github.com/usmannasir/cyberpanel/commit/f865d0cb027bf2e325be46c882e6ffc9f37896f7

**Comment by meramsey** [comment]
This is fixed in 2.1.1-dev now and once those fixes have been reviewed and copied to stable and 2.1.1 this can be closed.

https://github.com/usmannasir/cyberpanel/commit/f9cb98ef3e4c8af3adaff07718726afe79c35862
